packages/@aws-cdk/aws-rds/lib/cluster-engine.ts

```diff
@@ -0,0 +1,318 @@
+import { compare } from './private/version';
+
+/**
+ * The extra options passed to the {@link IClusterEngine.bindToCluster} method.
+ */
+export interface ClusterEngineBindOptions {
+  /**
+   * The role used for S3 importing.
+   *
+   * @default - none
+   */
+  readonly s3ImportRole?: iam.IRole;
+
+  /**
+   * The role used for S3 exporting.
+   *
+   *  @default - none
+   */
+  readonly s3ExportRole?: iam.IRole;
+
+  /**
+   * The customer-provided ParameterGroup.
+   *
+   * @default - none
+   */
+  readonly parameterGroup?: IParameterGroup;
+}
+
+/**
+ * The type returned from the {@link IClusterEngine.bindToCluster} method.
+ */
+export interface ClusterEngineConfig {
+  /**
+   * The ParameterGroup to use for the cluster.
+   *
+   * @default - no ParameterGroup will be used
+   */
+  readonly parameterGroup?: IParameterGroup;
+
+  /**
+   * The port to use for this cluster,
+   * unless the customer specified the port directly.
+   *
+   * @default - use the default port for clusters (3306)
+   */
+  readonly port?: number;
+}
+
+/**
+ * The interface representing a database cluster (as opposed to instance) engine.
+ */
+export interface IClusterEngine {
+  /** The type of the engine, for example "aurora-mysql". */
+  readonly engineType: string;
+
+  /**
+   * The exact version of a given engine.
+   *
+   * @default - default version for the given engine type
+   */
+  readonly engineVersion?: string;
+
+  /**
+   * The family to use for ParameterGroups using this engine.
+   * This is usually equal to "<engineType><engineMajorVersion>",
+   * but can sometimes be a variation of that.
+   * You can pass this property when creating new ClusterParameterGroup.
+   */
+  readonly parameterGroupFamily: string;
+
+  /** The application used by this engine to perform rotation for a single-user scenario. */
+  readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  /** The application used by this engine to perform rotation for a multi-user scenario. */
+  readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  /** The default EC2 instance type to use for databases in this cluster. */
+  readonly defaultInstanceType: ec2.InstanceType;
+
+  /**
+   * Method called when the engine is used to create a new cluster.
+   */
+  bindToCluster(scope: core.Construct, options: ClusterEngineBindOptions): ClusterEngineConfig;
+}
+
+interface BaseClusterEngineProps {
+  readonly engineType: string;
+  readonly needsS3RolesInParameters: boolean;
+  readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+  readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+  readonly parameterGroupFamilies?: ParameterGroupFamilyMapping[];
+  readonly defaultPort?: number;
+  readonly engineVersion?: string;
+}
+
+abstract class BaseClusterEngine implements IClusterEngine {
```

Comment:
We usually use `ClusterEngineBase`, i.e., `Base` as a suffix instead of prefix.

Reply:
Done.